lib/internal/modules/esm/translators.js

```diff
@@ -211,6 +221,7 @@ translators.set('commonjs', async function commonjsStrategy(url, source,
 });
 
 function cjsPreparseModuleExports(filename, source) {
+  // TODO: Do we want to keep hitting the user mutable CJS loader here?
```

Comment:
I think we need to in order to interoperate with transpiling workflows that overwrite require.extensions today, correct?

Reply:
I think there would be value in proposing an alternative that gets rid of all the legacy stuff (including `require.extensions`) – even if that can never be the default, because of backward compat – otherwise it's simply too difficult to integrate. Maybe it's a bad idea, we'll see, I'm not married to it, but I'd like us to try it.

Reply:
"Never" is quite a long time. Having a "no legacy" opt in would definitely be a very nice first step towards removing require.extensions eventually.